clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java

```diff
@@ -50,11 +53,29 @@ public void start() throws Exception {
         logger.log(LogLevel.Information, "WebSocket transport connected to: %s", webSocketClient.getURI());
     }
 
+    @Override
+    public CompletableFuture startAsync() throws Exception {
+        return CompletableFuture.runAsync(() -> {
```

Comment:
So I assume this is the equivalent of `Task.Run`... not ideal at all. I think maybe we should build a simple wrapper around the WebSocket client that uses futures. It looks like this should be straightforward enough. Instead of `connectBlocking`, we just create a `CompletableFuture`, call `connect` and return the future. In `onOpen` you complete the future.